### PR TITLE
Adds the ability to put prescription lenses in role-related optics

### PR DESCRIPTION
**Pull request by Staykeu**
## About the pull request

Adds the ability to apply lenses from prescription eyewear into HUDs/mesons/specialist optics.

## Explain why it's good for the game

QoL change that makes life easier for those that have characters with impaired vision, allows the use of these eyepieces without obscuring the mask slot, be it reserved for smokes, scarfs, coifs, etc.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

N/A

</details>

# Changelog
:cl: Stakeyng
add: Added the ability to combine prescription glasses with eyewear for varying roles
/:cl:

### CODE REVIEW

**Review by Drulikar — APPROVED** [review]
Code changes look fine (its code I wrote) and can confirm working in game.

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -87,6 +87,36 @@
 		if(istype(A, /datum/action/item_action/toggle))
 			A.update_button_icon()
 
+/obj/item/clothing/glasses/proc/try_make_offhand_prescription(mob/user)
+	if(!prescription)
+		return FALSE
+	if(QDELETED(src))
+		return FALSE
```
Comment (fira) [inline code/modules/clothing/glasses/glasses.dm:94]: That's redundant

**Review by fira — CHANGES_REQUESTED** [review]

File: code/modules/clothing/glasses/glasses.dm (hunk 1 above)
Comment (Staykeu) [reply code/modules/clothing/glasses/glasses.dm]: removed

**Review by fira — APPROVED** [review]

File: code/modules/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -87,6 +87,34 @@
...
 			A.update_button_icon()
 
+/obj/item/clothing/glasses/proc/try_make_offhand_prescription(mob/user)
+	if(!prescription)
+		return FALSE
+
+	var/obj/item/clothing/glasses/offhand = user.get_inactive_hand()
+	if(istype(offhand) && !offhand.prescription)
+		if(tgui_alert(user, "Do you wish to take out the prescription lenses and put them in [offhand]?", "Insert Prescription Lenses", list("Yes", "No")) == "Yes")
+			offhand.prescription = TRUE
```
Comment (Drulikar) [inline code/modules/clothing/glasses/glasses.dm:97]: I intended to have them guard against this, but I just didn't place it in the correct spot.
image

```suggestion
			if(QDELETED(src) || offhand != user.get_inactive_hand())
				return FALSE
			offhand.prescription = TRUE
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]
Code changes are fine.

**Review by fira — APPROVED** [review]

**Pull request merged**